x-pack/plugins/security/server/authorization/roles/elasticsearch_role.ts

```diff
@@ -184,6 +188,40 @@ function transformRoleApplicationsToKibanaPrivileges(
     };
   }
 
+  // if a feature privilege requires all spaces, but is assigned to other spaces, we won't transform these
```

Comment:
This logic still needs tests. I believe the place the other pieces of this function are tested is in `x-pack/plugins/security/server/routes/authorization/roles/get_all.test.ts`

Reply:
@criamico This should be the only TODO remaining, pending any additional PR comments.

Reply:
Thanks for the additional context! I'll add these tests.